Markdown Reference Page

DFriend01 opened the pull request:
Resolves #36.

## What is Included
I have included the following topics:
- What is Markdown?
- Getting Started
- Where and how we use Markdown on Sailbot

I also linked to the necessary resources for tutorials on Markdown and references for the different flavors of Markdown that we use.

## Other Comments

### Markdown Linting and Checks
I chose **not** to include a section on the `.json` files in the repository related to Markdown. I feel like these topics are more appropriate for either the section on MkDocs or VS Code. Let me know what your thoughts are.

### VS Code Extensions
I briefly considered talking about some helpful VS Code extensions for Markdown, but I was not sure if it was particularly relevant. We also don't use extensions for Markdown in the Sailbot Workspace if I recall correctly. What do you think? I was thinking of the following extensions:
- [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)
- [Markdown Preview GitHub Styling](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-preview-github-styles)

DFriend01 committed:
Starting implementation of the Markdown reference page

patrick-5546 commented on docs/reference/markdown/markdown.md:
```diff
@@ -0,0 +1,71 @@
+# Markdown
+
+Markdown is a lightweight markup language that you can use to add formatting elements to plaintext text documents[^1].
```
Footnote after punctuation

patrick-5546 commented on docs/reference/markdown/markdown.md:
```diff
@@ -0,0 +1,71 @@
+Almost anywhere in GitHub where you can write text, you can format it with Markdown. GitHub also allows you to preview
+your Markdown before you submit any comments.
+
+=== "Before Rendering"
+
+    ![image](../../assets/images/markdown/issue_md_text.png)
+
+=== "After Rendering"
+
+    ![image](../../assets/images/markdown/issue_md_rendered.png)
```
Fancy, I like it!

DFriend01 replied:
$T_HAN^KS$

patrick-5546 commented on docs/reference/markdown/markdown.md:
```diff
@@ -0,0 +1,71 @@
+even further with HTML/CSS)!
+
+Material for MkDocs supports most of the common Markdown syntax. However, it also adds on to its flavour of Markdown
+by adding many of its own customized features. Feel free to look at the [source code](https://github.com/UBCSailbot/docs)
+for this site for how our pages were generated using MkDocs-flavoured Markdown.
```
Consider linking to the `docs/` folder, where our site's pages are contained

DFriend01 replied:
Done

patrick-5546 commented on docs/reference/markdown/markdown.md:
```diff
@@ -0,0 +1,71 @@
+by adding many of its own customized features. Feel free to look at the [source code](https://github.com/UBCSailbot/docs)
+for this site for how our pages were generated using MkDocs-flavoured Markdown.
+
+!!! note "MkDocs-Flavoured Markdown"
```
Material for MkDocs, what we use, is a subset of MkDocs. It makes it much quicker and easier to create beautiful documentation.

patrick-5546 commented on docs/reference/markdown/markdown.md:
```diff
@@ -0,0 +1,71 @@
+
+!!! note "MkDocs-Flavoured Markdown"
+
+    Many of the features of MkDocs can be utilized through its flavour of Markdown. Refer the the 
```
Consider changing first sentence to something akin to:

Material for MkDocs' flavour of Markdown extends upon vanilla Markdown, adding features that can be used to take your technical documentation to the next level.

patrick-5546 reviewed (commented):
Sorry for the spam, I forgot to add comments to a review.

It looks great! I'm thinking we can close out #27 (link it to this PR), I don't think we need a separate page to go over Material for MkDocs considering its official documentation is great and it's intuitive to learn (especially since we have pages in our repo to refer to).

Addressing your PR description comments:
- I think linting, including markdown, should go in the vscode page
- I don't like all in one, but I haven't seen the GitHub preview before and it looks amazing. I added it to Eric's PR's in docs and sailbot_workspace that update our vscode extensions

DFriend01 commented:
We already link to the documentation for Material for MkDocs. It is probably fair to say that this PR also resolves #27.

DFriend01 committed:
Addressed PR comments

DFriend01 committed:
Fixed broken link

DFriend01 committed:
Linting

patrick-5546 committed:
Make docs directory structure consistent

patrick-5546 committed:
Proofread

patrick-5546 reviewed (commented):
I changed my mind, I think we should have a section on linting to match our Python and C++ documentation. It also might be helpful to go over different methods of rendering Markdown locally. In the VS Code page I will recommend specific extensions for each language.

Notes:
- I forced pushed to remove the infrastructure changes, putting them in #71 instead
- Both GitHub and Material docs fully support vanilla Markdown, they only add/extend functionally

DFriend01 committed:
Added section on rendering Markdown locally

DFriend01 committed:
Added section on Markdown linting

DFriend01 commented:
@patrick-5546 I added the following additional sections to the Markdown page:
- Rendering Markdown Locally
- Markdown Linting

Also, the CI seems to be failing on a working link. I am thinking of adding that link to [.markdown-link-check.json](https://github.com/UBCSailbot/docs/blob/main/.markdown-link-check.json) under the ignore list.

UPDATE: I added an ignore pattern to ignore the GitHub docs links.

DFriend01 committed:
Add to markdown link check to ignore github docs

patrick-5546 committed:
Proofread

patrick-5546 commented on docs/reference/markdown.md:
```diff
@@ -0,0 +1,94 @@
+    [official Material for MkDocs reference page](https://squidfunk.github.io/mkdocs-material/reference/) for more
+    information on the available features.
+
+## Rendering Markdown Locally
+
+You have a few choices to render Markdown on your computer.
+Be advised that if you are using an extended version of Markdown, you will
+need to consult the documentation from the service provider to render their flavour of Markdown properly. The following
+resources are good for rendering vanilla Markdown:
```
I'm thinking we can this section to just "Rendering Markdown"
- Vanilla markdown options are good
- For GitHub, recommend drafting an issue or something and the GitHub preview VS Code extension
- For mkdocs, recommend running the site locally  with `mkdocs serve`

DFriend01 committed:
Updated Markdown Rendering section

patrick-5546 committed:
Update rendering material for mkdocs

patrick-5546 reviewed (approved):
Great work on this PR, feel free to merge
